Source/Core/Common/StringUtil.cpp

```diff
@@ -197,26 +197,42 @@ std::string StringFromFormatV(const char* format, va_list args)
   return temp;
 }
 
-// For Debugging. Read out an u8 array.
-std::string ArrayToString(const u8* data, u32 size, int line_len, bool spaces)
+// This assumes you already masked the upper four bits.
+static constexpr char NibbleToHex(const u8 n)
 {
-  std::ostringstream oss;
-  oss << std::setfill('0') << std::hex;
+  if (n < 0xa)
```

Comment:
Can you put an `assert(n <= 0xf)` or similar in here? You'll also need to include `<cassert>`.